src/util/logging.py

```diff
@@ -17,30 +17,44 @@
 
 from util.context import Context
 
+log_call_count = dict()
+
+
+def log_multiline_message(message: Text, caller: Text):
+
+    # display logs called from one spot no more than 10 times, "None" means it was counted before
+    if caller != "None":
+        if log_call_count.get(caller, 0) < 10:
+            log_call_count[caller] = log_call_count.get(caller, 0) + 1
```

Comment:
here, if we are at last possible occurence (lets say limit is 10 and this is 10th call) then I think we should add log entry "Logs for caller X have reached throttling limit of Y, further logs from this caller will be discarded" 

like: 
if calls < 10 - bump dict
if calls == 10 - print warning and return
if calls > 10 - return